tests/test_ufuncs.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+import fast_numpy_loops
+import pytest
+fast_numpy_loops.initialize()
+
+
+rng = np.random.default_rng(123456)
+
+def type2dtype(types):
+    """
+    Maps the ufunc.type to the input, output dtypes.
+    type2dtype('ii->?') -> (np.int32, np.int32), (np.bool_,)
+    """
+    inp, out = types.split('->')
+    return tuple(np.dtype(c) for c in inp), tuple(np.dtype(c) for c in out)
+
+def get_ufuncs_and_types():
+    """Create a dictionary with keys of ufunc names and values of all the
+    supported type signatures
+    """
+    ufuncs = [x for x in dir(np) if isinstance(getattr(np, x), np.ufunc)]
+    # Maybe use a collections.defaultdict instead?
+    ret = dict([[x, []] for x in ufuncs])
+    for s in ret:
+        ret[s] = [type2dtype(t) for t in getattr(np, s).types]
+    return ret
+
+def fill_random(a):
+    """Fill an ndarray with random values. This will uniformly cover the
+    bit-valued number space, which in the case of floats is differnt from
+    rng.uniform()
+    """
+    if a.dtype == 'object':
+        v = a.reshape(-1)
+        # Slow !!!
+        for i in range(v.size):
+            v[i] = float(rng._bit_generator.random_raw(1))
+    else:
+        v = a.view(np.uint64)
+        v[:] = rng._bit_generator.random_raw(v.size).reshape(v.shape)
+    return a
+
+typemap = get_ufuncs_and_types()
+
+def data(in_dtypes, out_dtypes, shape):
+    """ Return two tuples: input and output, with random data dtypes and
+    shape
+    """
+    ret_in = [fill_random(np.empty(shape, dtype=d)) for d in in_dtypes]
+    ret_out = tuple([fill_random(np.empty(shape, dtype=d)) for d in out_dtypes])
+    return ret_in, ret_out
+
+@pytest.mark.filterwarnings("ignore::RuntimeWarning")
+@pytest.mark.parametrize(['name', 'types'], ([k, v] for k,v in typemap.items()))
+def test_threads(name, types):
+    """ Test that enabling the threading does not change the results
+    """
+    ufunc = getattr(np, name)
+    for in_dtypes, out_dtypes in types:
+        # Skip object dtypes
```

Comment:
was this skipped because of the slow value generation above? If so then it can perhaps be enabled now

Reply:
No, object dtypes were skipped because they are annoying to benchmark: they are much slower than int/float types, and cannot be optimized for.